<?xml version='1.0' encoding='utf-8'?>
<sdf version="1.5">
  <world name="default">
    <include>
      <uri>model://sun</uri>
    </include>
    <include>
      <uri>model://ground_plane</uri>
      <pose>0 0 -0.001 0 0 0</pose>
    </include>
    <model name="airspace">
      <link name="link_0">
        <pose frame="">0 0 2.5 0 0 0</pose>
        <gravity>0</gravity>
        <self_collide>0</self_collide>
        <kinematic>1</kinematic>
        <visual name="airspace_visual">
          <pose frame="">0 0 0 0 0 0</pose>
          <geometry>
            <box>
              <size>20 20 5.0</size>
            </box>
          </geometry>
          <material>
            <script>
              <uri>file://media/materials/scripts/gazebo.material</uri>
              <name>Gazebo/DarkOrangeTransparentOverlay</name>
            </script>
          </material>
          <cast_shadows>0</cast_shadows>
          <transparency>0.75</transparency>
        </visual>
      </link>
    </model>
    <include>
      <uri>model://aruco_visual_marker_1_pad</uri>
      <pose>0.5 -0.5 0.0 0.0 0.0 0.785398163397</pose>
    </include>
    <include>
      <uri>model://aruco_visual_marker_2_pad</uri>
      <pose>0.5 0.5 0.0 0.0 0.0 2.35619449019</pose>
    </include>
    <include>
      <uri>model://aruco_visual_marker_3_pad</uri>
      <pose>-0.5 0.5 0.0 0.0 0.0 -2.35619449019</pose>
    </include>
    <include>
      <uri>model://aruco_visual_marker_4_pad</uri>
      <pose>-0.5 -0.5 0.0 0.0 0.0 -0.785398163397</pose>
    </include>
    <include>
      <uri>model://gate</uri>
      <pose>0.5 -0.5 0.2 0 0 2.35619449019</pose>
    </include>
    <model name="wall_0">
      <link name="link_0">
        <pose frame="">1.25 1.5 2.5 0 0 2.26553460299</pose>
        <gravity>0</gravity>
        <self_collide>0</self_collide>
        <kinematic>1</kinematic>
        <visual name="wall_0_visual">
          <pose frame="">0 0 0 0 0 0</pose>
          <geometry>
            <box>
              <size>3.90512483795 0.01 5.0</size>
            </box>
          </geometry>
          <material>
            <script>
              <uri>file://media/materials/scripts/gazebo.material</uri>
              <name>Gazebo/DarkOrangeTransparentOverlay</name>
            </script>
          </material>
          <cast_shadows>1</cast_shadows>
          <transparency>0</transparency>
        </visual>
      </link>
    </model>
    <include>
      <uri>model://sign_stop</uri>
      <pose>2.5 -2.5 0.0 0.0 0.0 0.785398163397</pose>
    </include>
    <physics default="0" name="default_physics" type="ode">
      <gravity>0 0 -9.8066</gravity>
      <ode>
        <solver>
          <type>quick</type>
          <iters>20</iters>
          <sor>1.3</sor>
          <use_dynamic_moi_rescaling>0</use_dynamic_moi_rescaling>
        </solver>
        <constraints>
          <cfm>0</cfm>
          <erp>0.2</erp>
          <contact_max_correcting_vel>100</contact_max_correcting_vel>
          <contact_surface_layer>0.001</contact_surface_layer>
        </constraints>
      </ode>
      <max_step_size>0.002</max_step_size>
      <real_time_factor>1</real_time_factor>
      <real_time_update_rate>500</real_time_update_rate>
      <magnetic_field>6e-06 2.3e-05 -4.2e-05</magnetic_field>
    </physics>
  </world>
</sdf>
